discovery-provider/src/solana/audius_data_transaction_handlers.py

```diff
@@ -381,6 +415,63 @@ def update_user_model_metadata(
         user_record.creator_node_endpoint = metadata_dict["creator_node_endpoint"]
 
 
+def update_track_model_metadata(
+    session: Session, track_record: Track, track_metadata: Dict
+):
+    track_record.title = track_metadata["title"]
+    track_record.length = track_metadata["length"] or 0
+    track_record.cover_art_sizes = track_metadata["cover_art_sizes"]
+    if track_metadata["cover_art"]:
+        track_record.cover_art_sizes = track_record.cover_art
+        # TODO check if blacklisted?
```

Comment:
yep, definitely need this as well

Reply:
i'm adding the upstream fetch for existing records